src/createBaseForm.js

```diff
@@ -45,9 +45,9 @@ function createBaseForm(option = {}, mixins = []) {
         };
       },
 
-      componentWillReceiveProps(nextProps) {
+      componentWillReceiveProps(nextProps, nextContext) {
         if (mapPropsToFields) {
-          this.fields = mapPropsToFields(nextProps);
+          this.fields = mapPropsToFields(nextProps, nextContext);
```

Comment:
I think you can use a HOC to convert context to props, and then pass it to form.